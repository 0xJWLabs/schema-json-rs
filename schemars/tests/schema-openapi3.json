{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Schema",
  "anyOf": [
    {
      "type": "boolean"
    },
    {
      "$ref": "#/components/schemas/SchemaRef"
    },
    {
      "$ref": "#/components/schemas/SchemaObject"
    }
  ],
  "definitions": {
    "InstanceType": {
      "enum": [
        "null",
        "boolean",
        "object",
        "array",
        "number",
        "string",
        "integer"
      ]
    },
    "Schema": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaRef"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ]
    },
    "SchemaObject": {
      "type": "object",
      "properties": {
        "$id": {
          "type": "string",
          "nullable": true
        },
        "$schema": {
          "type": "string",
          "nullable": true
        },
        "allOf": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          },
          "nullable": true
        },
        "anyOf": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          },
          "nullable": true
        },
        "definitions": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/components/schemas/Schema"
          }
        },
        "description": {
          "type": "string",
          "nullable": true
        },
        "enum": {
          "type": "array",
          "items": {},
          "nullable": true
        },
        "items": {
          "anyOf": [
            {
              "$ref": "#/components/schemas/Schema"
            },
            {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/Schema"
              }
            }
          ],
          "nullable": true
        },
        "not": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaRef"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "oneOf": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          },
          "nullable": true
        },
        "properties": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/components/schemas/Schema"
          }
        },
        "required": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "title": {
          "type": "string",
          "nullable": true
        },
        "type": {
          "anyOf": [
            {
              "$ref": "#/components/schemas/InstanceType"
            },
            {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/InstanceType"
              }
            }
          ],
          "nullable": true
        }
      },
      "additionalProperties": true
    },
    "SchemaRef": {
      "type": "object",
      "properties": {
        "$ref": {
          "type": "string"
        }
      }
    },
    "SingleOrVec_For_InstanceType": {
      "anyOf": [
        {
          "$ref": "#/components/schemas/InstanceType"
        },
        {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/InstanceType"
          }
        }
      ]
    },
    "SingleOrVec_For_Schema": {
      "anyOf": [
        {
          "$ref": "#/components/schemas/Schema"
        },
        {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          }
        }
      ]
    }
  }
}