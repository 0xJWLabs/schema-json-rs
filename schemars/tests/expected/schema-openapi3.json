{
  "$schema": "https://spec.openapis.org/oas/3.0/schema/2019-04-02#/definitions/Schema",
  "title": "RootSchema",
  "type": "object",
  "properties": {
    "$id": {
      "type": "string",
      "nullable": true
    },
    "$ref": {
      "type": "string",
      "nullable": true
    },
    "$schema": {
      "type": "string",
      "nullable": true
    },
    "additionalItems": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "additionalProperties": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "allOf": {
      "type": "array",
      "items": {
        "$ref": "#/components/schemas/Schema"
      },
      "nullable": true
    },
    "anyOf": {
      "type": "array",
      "items": {
        "$ref": "#/components/schemas/Schema"
      },
      "nullable": true
    },
    "const": {
      "nullable": true
    },
    "contains": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "default": {
      "nullable": true
    },
    "definitions": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/components/schemas/Schema"
      }
    },
    "deprecated": {
      "type": "boolean"
    },
    "description": {
      "type": "string",
      "nullable": true
    },
    "else": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "enum": {
      "type": "array",
      "items": {},
      "nullable": true
    },
    "exclusiveMaximum": {
      "type": "number",
      "format": "double",
      "nullable": true
    },
    "exclusiveMinimum": {
      "type": "number",
      "format": "double",
      "nullable": true
    },
    "format": {
      "type": "string",
      "nullable": true
    },
    "if": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "items": {
      "anyOf": [
        {
          "$ref": "#/components/schemas/Schema"
        },
        {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          }
        }
      ],
      "nullable": true
    },
    "maxItems": {
      "type": "integer",
      "format": "uint32",
      "minimum": 0.0,
      "nullable": true
    },
    "maxLength": {
      "type": "integer",
      "format": "uint32",
      "minimum": 0.0,
      "nullable": true
    },
    "maxProperties": {
      "type": "integer",
      "format": "uint32",
      "minimum": 0.0,
      "nullable": true
    },
    "maximum": {
      "type": "number",
      "format": "double",
      "nullable": true
    },
    "minItems": {
      "type": "integer",
      "format": "uint32",
      "minimum": 0.0,
      "nullable": true
    },
    "minLength": {
      "type": "integer",
      "format": "uint32",
      "minimum": 0.0,
      "nullable": true
    },
    "minProperties": {
      "type": "integer",
      "format": "uint32",
      "minimum": 0.0,
      "nullable": true
    },
    "minimum": {
      "type": "number",
      "format": "double",
      "nullable": true
    },
    "multipleOf": {
      "type": "number",
      "format": "double",
      "nullable": true
    },
    "not": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "oneOf": {
      "type": "array",
      "items": {
        "$ref": "#/components/schemas/Schema"
      },
      "nullable": true
    },
    "pattern": {
      "type": "string",
      "nullable": true
    },
    "patternProperties": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/components/schemas/Schema"
      }
    },
    "properties": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/components/schemas/Schema"
      }
    },
    "propertyNames": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "readOnly": {
      "type": "boolean"
    },
    "required": {
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    "then": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ],
      "nullable": true
    },
    "title": {
      "type": "string",
      "nullable": true
    },
    "type": {
      "anyOf": [
        {
          "$ref": "#/components/schemas/InstanceType"
        },
        {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/InstanceType"
          }
        }
      ],
      "nullable": true
    },
    "uniqueItems": {
      "type": "boolean",
      "nullable": true
    },
    "writeOnly": {
      "type": "boolean"
    }
  },
  "additionalProperties": true,
  "definitions": {
    "InstanceType": {
      "enum": [
        "null",
        "boolean",
        "object",
        "array",
        "number",
        "string",
        "integer"
      ]
    },
    "Schema": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/components/schemas/SchemaObject"
        }
      ]
    },
    "SchemaObject": {
      "type": "object",
      "properties": {
        "$id": {
          "type": "string",
          "nullable": true
        },
        "$ref": {
          "type": "string",
          "nullable": true
        },
        "additionalItems": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "additionalProperties": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "allOf": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          },
          "nullable": true
        },
        "anyOf": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          },
          "nullable": true
        },
        "const": {
          "nullable": true
        },
        "contains": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "default": {
          "nullable": true
        },
        "deprecated": {
          "type": "boolean"
        },
        "description": {
          "type": "string",
          "nullable": true
        },
        "else": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "enum": {
          "type": "array",
          "items": {},
          "nullable": true
        },
        "exclusiveMaximum": {
          "type": "number",
          "format": "double",
          "nullable": true
        },
        "exclusiveMinimum": {
          "type": "number",
          "format": "double",
          "nullable": true
        },
        "format": {
          "type": "string",
          "nullable": true
        },
        "if": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "items": {
          "anyOf": [
            {
              "$ref": "#/components/schemas/Schema"
            },
            {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/Schema"
              }
            }
          ],
          "nullable": true
        },
        "maxItems": {
          "type": "integer",
          "format": "uint32",
          "minimum": 0.0,
          "nullable": true
        },
        "maxLength": {
          "type": "integer",
          "format": "uint32",
          "minimum": 0.0,
          "nullable": true
        },
        "maxProperties": {
          "type": "integer",
          "format": "uint32",
          "minimum": 0.0,
          "nullable": true
        },
        "maximum": {
          "type": "number",
          "format": "double",
          "nullable": true
        },
        "minItems": {
          "type": "integer",
          "format": "uint32",
          "minimum": 0.0,
          "nullable": true
        },
        "minLength": {
          "type": "integer",
          "format": "uint32",
          "minimum": 0.0,
          "nullable": true
        },
        "minProperties": {
          "type": "integer",
          "format": "uint32",
          "minimum": 0.0,
          "nullable": true
        },
        "minimum": {
          "type": "number",
          "format": "double",
          "nullable": true
        },
        "multipleOf": {
          "type": "number",
          "format": "double",
          "nullable": true
        },
        "not": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "oneOf": {
          "type": "array",
          "items": {
            "$ref": "#/components/schemas/Schema"
          },
          "nullable": true
        },
        "pattern": {
          "type": "string",
          "nullable": true
        },
        "patternProperties": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/components/schemas/Schema"
          }
        },
        "properties": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/components/schemas/Schema"
          }
        },
        "propertyNames": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "readOnly": {
          "type": "boolean"
        },
        "required": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "then": {
          "anyOf": [
            {
              "type": "boolean"
            },
            {
              "$ref": "#/components/schemas/SchemaObject"
            }
          ],
          "nullable": true
        },
        "title": {
          "type": "string",
          "nullable": true
        },
        "type": {
          "anyOf": [
            {
              "$ref": "#/components/schemas/InstanceType"
            },
            {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/InstanceType"
              }
            }
          ],
          "nullable": true
        },
        "uniqueItems": {
          "type": "boolean",
          "nullable": true
        },
        "writeOnly": {
          "type": "boolean"
        }
      },
      "additionalProperties": true
    }
  }
}